### ISSUE TITLE
Add `tsc` to linting step

### ISSUE THREAD

**Issue by rijkvanzanten** [directus/directus#7953]
As @Oreilles correctly pointed out, `vite` doesn't seem to type check the code at the moment. We should update the `lint` command (and therefore the lint CI step) to run `tsc` globally to make sure there's no type regressions introduced

**Comment by Oreilles** [comment]
Linting don't check for type error in any of the packages so I don't think adding it to the `app` is the way to go.
Type checking currently only happens during `npm run dev` and `npm run build`, it makes sense to keep it that way.
Suggestion for `app/package.json`:

```diff
- "dev": "vite",
- "build": "vite build",
+ "dev": "run-p dev:*",
+ "dev:tsc": "tsc -w --noEmit",
+ "dev:vite": "vite",
+ "build": "tsc --noEmit && vite build",
```

**Comment by rijkvanzanten** [comment]
> Linting don't check for type error in any of the packages so I don't think adding it to the app is the way to go.

True! I was thinking we'd add it _globally_ on the root of the repo, so we check _every_ bit of TS in the lint step

**Comment by nickrum** [comment]
We'd have to use something like [vue-tsc](https://github.com/johnsoncodehk/vue-tsc) to be able to type-check `*.vue` files.

Doing type-checking as part of the lint step (making it run on each commit) might be a bit heavy.

**Comment by Oreilles** [comment]
Agree with @nickrum, linting step should just do linting. Type checking should stay part of the dev/build step imo.

**Comment by rijkvanzanten** [comment]
> Type checking should stay part of the dev/build step imo.

Interesting, I always considered type checking as an additional safety to ensure code quality, not necessarily a build requirement 🤔

**Comment by Oreilles** [comment]
If anything type checking feels closer to testing than linting. But it seems that both are accepted. [slate](https://github.com/ianstormtaylor/slate/blob/main/package.json) and [elastic](https://github.com/elastic/eui/blob/master/package.json) does it during linting, [rxjs](https://github.com/ReactiveX/rxjs/blob/master/package.json) and [vue](https://github.com/vuejs/vue-next/blob/master/package.json) during testing.
